### PR TITLE
Rewrite resource with powershell cmdlets

### PR DESCRIPTION
**Pull request by Happycoil**
Signed-off-by: Tor Magnus Rakvåg <tm@intility.no>

### Description

Rewrites the cookbook to use powershell cmdlets instead of netsh.

### Issues Resolved

https://github.com/chef-cookbooks/windows_firewall/issues/9
https://github.com/chef-cookbooks/windows_firewall/issues/23

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: resources/rule.rb [hunk 1]
```diff
@@ -17,40 +17,62 @@
...
-property :firewall_action, Symbol, default: :allow, equal_to: [:allow, :block, :bypass]
-property :profile, Symbol, default: :any, equal_to: [:public, :private, :domain, :any]
+property :firewall_action, Symbol, default: :allow, equal_to: [:allow, :block, :notconfigured],
+          coerce: proc { |f| f.is_a?(String) ? f.downcase.to_sym : f }
+property :profile, Symbol, default: :any, equal_to: [:public, :private, :domain, :any, :notapplicable],
+          coerce: proc { |p| p.is_a?(String) ? p.downcase.to_sym : p }
 property :program, String
 property :service, String
-property :interfacetype, Symbol, default: :any, equal_to: [:any, :wireless, :lan, :ras]
+property :interface_type, Symbol, default: :any, equal_to: [:any, :wireless, :wired, :remoteaccess],
```
Comment (tas50) [inline resources/rule.rb]: If you want to allow strings and then coerce them to symbols you need to include String as an allowed type

File: resources/rule.rb [hunk 2]
```diff
@@ -17,40 +17,62 @@
...
 load_current_value do
-  current_rule = shell_out("chcp 437 >null 2>&1 && cmd.exe /c netsh advfirewall firewall show rule name=\"#{rule_name}\"")
-
-  current_value_does_not_exist! if current_rule.stdout.strip == 'No rules match the specified criteria.'
+  load_state_cmd = load_firewall_state(rule_name)
+  output = powershell_out(load_state_cmd)
+  if output.stdout.empty?
+    current_value_does_not_exist!
+  else
+    state = Chef::JSONCompat.from_json(output.stdout)
```
Comment (tas50) [inline resources/rule.rb]: You should probably require the json compat module. It's coming in due to other things requiring it, but that might break some day

Comment (Happycoil) [reply resources/rule.rb]: fixed

**Comment by tas50** [comment]
Merging so I can try to get this working with appveyor

**Pull request merged**